compaction/compaction_manager.cc

```diff
@@ -842,12 +849,15 @@ future<> compaction_manager::drain() {
 }
 
 future<> compaction_manager::stop() {
+    if (_compaction_module) {
+        co_await std::exchange(_compaction_module, nullptr)->stop();
+    }
     // never started
```

Comment:
it's more idiomatic to write it like this:

if (auto cm = std::exchange(_compaction_module, nullptr)) { co_await cm->stop(); }